lib/src/clojure_lsp/settings.clj

```diff
@@ -76,11 +75,11 @@
 (defn all
   "Get memoized settings from db.
   Refreshes settings if memoize threshold met."
-  [db]
```

Comment:
Passing the derefed db to `settings/all` and `setting/get` reveals many other places that don't need the atom. I was careful to test that the memoization works the same... The settings are cached for 1 second, after which the settings file is re-read, to check for changes. This can still lead to timing bugs, but only if the user is updating their settings file. Still better would be if the settings were calculated less frequently, much higher in the call hierarchy, so that all main entrypoints could rely on having a consistent view of the settings. But that's a separate project.

Reply:
yes, sounds good enough to me this way for now, nice!